src/lib/fcitx/userinterface.h

```diff
@@ -46,6 +46,20 @@ class FCITXCORE_EXPORT UserInterface : public AddonInstance {
     virtual bool available() = 0;
     virtual void suspend() = 0;
     virtual void resume() = 0;
+    virtual bool isVirtualKeyboard();
```

Comment:
I think this should be bool isVirtualKeyboard const(). But put that aside, if no addon will support both at the same time, I'd suggest add a new field in AddonInfo instead. So we know before the loading happens.

Reply:
In your opinion, what's the right class to add the new field? class AddonInfoPrivate or class AddonConfig？ And what's the name of the new field?

Reply:
See how we have a field dedicated for UI Addon: https://github.com/fcitx/fcitx5/blob/08e0a5f9367466fb4360d0a8c2dbc80e9406d9ff/src/lib/fcitx/addoninfo.cpp#L26

You can add what ever you think that makes sense. for me, in order to allow flexibility, maybe a enum type "UIType" instead of bool.